src/Nethermind.Verkle/Proofs/MultiProof.cs

```diff
@@ -118,20 +118,38 @@ public bool CheckMultiProof(Transcript transcript, VerkleVerifierQuery[] queries
         transcript.AppendPoint(d, "D");
         FrE t = transcript.ChallengeScalar("t");
 
-        FrE[] g2Den = new FrE[queries.Length];
+        FrE[] groupedEvals = new FrE[domainSize];
+        // Calculate groupedEvals = r * y_i.
         for (int i = 0; i < queries.Length; i++)
         {
-            g2Den[i] = t - FrE.SetElement(u0: queries[i].ChildIndex);
+            groupedEvals[queries[i].ChildIndex] += powersOfR[i] * queries[i].ChildHash;
         }
```

Comment:
Since we can have multiple queries targeting the same evaluation point (i.e: `queries[i].ChildIndex`), we first aggregate those evaluations in `groupedEvals`. And then we'll only work with this.

This shreds work when calculating inverses below.